.github/workflows/auto_release.yaml

```diff
@@ -0,0 +1,71 @@
+name: auto_release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  release:
+    name: Publish GitHub release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Fetch all history for all tags and branches
+        run: git fetch --prune --unshallow
+      - name: Install gcc-mingw-w64-x86-64
+        run: |
+             sudo apt -q update
+             sudo apt -q install --no-install-recommends gcc-mingw-w64-x86-64
+      - name: Build truckersmp-cli.exe
+        run: make
+      - name: Strip truckersmp-cli.exe
+        run: x86_64-w64-mingw32-strip truckersmp-cli.exe
+      - name: Compress files
+        id: compress_files
+        run: |
+             tag="$(basename ${{ github.ref }})"
+             tardir="$(basename ${{ github.repository }})-${tag:1}"
+             mkdir -p .output/$tardir
+             mv * .output/$tardir
+             tar Jcvf output.tar.xz -C .output --owner=root --group=root $tardir
+             echo "::set-output name=filename::${tardir}.tar.xz"
+             echo "::set-output name=version::${tag:1}"
+      - name: Generate changelog
+        id: generate_changelog
+        run: |
+             import subprocess as subproc
+             args = ("git", "tag", "-l", "--sort=-v:refname")
+             basetag = subproc.check_output(args).decode("utf-8").splitlines()[1]
+             args = (
+               "git", "log",
+               "--format= * [%h] %s", "--no-merges", "--reverse", basetag + "..",
+             )
+             output = subproc.check_output(args).decode("utf-8")
+             body = ""
+             if output:
+                 body += "## Changes since {}%0A".format(basetag[1:])
```

Comment:
Not sure about stripping the `v`.
We haven't decided of the version naming yet and you've tested with a date where this makes sense.
But with normal version number (which I'm generally more a fan of) this somehow looks a bit strange. Not sure, what do you think?

## Changes since 3.8

Reply:
The expected tag naming has been changed. (`v20.5` -> `20.5`)

Regarding the version naming, I would like to use version numbers that *indicate release year* (and month or release number) because our software needs to be frequently updated (so far) for frequent TruckersMP/Steam/Proton/Games changes.

* 2020.05 (May 2020)
* 20.05 (May 2020)
* 2020.5 (May 2020)
* 20.5 (May 2020)
* 2020.0 (1st release in 2020)
* 20.0 (1st release in 2020)

I think `20.5` format would be simple and easy to compare.

Reply:
> The expected tag naming has been changed. (`v20.5` -> `20.5`)

Maybe I was unclear, the prepended `v` in the tag is fine - it seems often used by other projects.

> I would like to use version numbers that _indicate release year_ (and month or release number) because our software needs to be frequently updated (so far) for frequent TruckersMP/Steam/Proton/Games changes.

I don't get your point.
* Our software won't have to be released frequently, instead we have to adapt quickly so the mod stays playable. Besides Protons base version is everything on auto-update already and a new Proton version came twice a year. And thanks to the `proton.json` we don't have to release for that case either.
* Even if we would have to release frequently a year-month scheme wouldn't make sense. Assuming it's frequently there would be multiple releases per month so how would the numbering be? `20.5.3`? `2020.05.3`? Looks like the third day in may.
* I'm assuming right now that once we're done with the initial work we don't have to release quite often. So date based versions can look outdated after some time even if they aren't.

Reply:
> Maybe I was unclear, the prepended v in the tag is fine - it seems often used by other projects. My comment was about the visible part in the release note section.

* "Changes since *v*0.1.2": It might be OK but I like "since 0.1.2" format. Wine announcement pages use "Changes since [version]" format.
* "Release *v*0.1.2": "Release v" is a bit strange for me.

Non "v" prefixed tag naming makes .yaml file and asset URLs simpler. I would like to use the naming unless there are problems.

Regarding the version naming, I think it's OK to use "normal" version numbers.
What do you think about the initial version number? Is `0.1.0` good?

> Note: This job won't work yet because there are no tags.

What do you think of setting initial tag to past commit (e.g. `git tag 0.0.1 8e95826`) ?

Reply:
> Regarding the version naming, I think it's OK to use "normal" version numbers.

I'm not generally against date versions. I've described my argument against that version scheme above, since you'd wanted to use them, do you have an argument why we should use them?

> setting initial tag to past commit

I would tag 8aa88ca as the first tagged initial version.

> What do you think about the initial version number?

Yes, `0.1.0` seems good for the initial tag. When we're done with the initial things we want to have included we can tag to `1.0.0`. If you think there's something tag-worthy in between then we can also tag additional minor versions.

Reply:
> since you'd wanted to use them, do you have an argument why we should use them?

I thought that user-friendly versioning for our users is indicating when it was released.

> instead we have to adapt quickly so the mod stays playable.

Sorry, "frequency" was not the point. but I think that the line you wrote can be one of the reasons why we use year-based versioning.

However, I also think that we only need to "adapt quickly" and versioning is not related to it.

> Even if we would have to release frequently a year-month scheme wouldn't make sense. Assuming it's frequently there would be multiple releases per month so how would the numbering be? 20.5.3? 2020.05.3? Looks like the third day in may.

I think the versions don't need to contain month because (I guess) we are not going to publish releases once a month: `2020.0` or `20.0` for 1st release in 2020, next `2020.1`/`20.1`, `2020.2`/`20.2`, ...

Reply:
One selling point of semver is that we could clearly indicate a breaking change with the major version. But right now I don't think we're going to have that.

When you're arguing with user friendliness then we have to go with the full year and begin counting at 1.

Let's do `2020.0` for the initial tag needed for this PR to work and continue with `2020.1`, `2020.2` … `2021.1`.

Reply:
> One selling point of semver is that we could clearly indicate a breaking change with the major version. But right now I don't think we're going to have that.

Yes. Since our software is essentially a launcher, I can hardly imagine "breaking changes".

> When you're arguing with user friendliness then we have to go with the full year and begin counting at 1.

1? "[year].[**0**, 1, 2, ...]" versioning is being used by multiple projects:

* mesa3d.org
* github.com/uglide/RedisDesktopManager/releases
* github.com/IntelPython/container-images/releases
* github.com/att/ast/releases

> Let's do `2020.0` for the initial tag needed for this PR to work and continue with `2020.1`, `2020.2` … `2021.1`.

Let's do `2020.0` (8aa88ca is OK) for the initial tag needed for this PR to work and continue with `2020.1`, `2020.2` … `2021.0`.

Reply:
> is being used by multiple projects

All your examples aren't targeting "Users", that's libraries and container software which is used by developers.
I have an example for an end-user application (game) which starts by 1: https://github.com/unknown-horizons/unknown-horizons/releases
Sadly that's also the only one I remember using the date format.

One primary argument for the date format is user friendliness - do (end-)users start counting at 0 for a first release in 2021?

Reply:
> One primary argument for the date format is user friendliness - do (end-)users start counting at 0 for a first release in 2021?

All I can say is "Some people would think so, and others would not think so." I feel that starting counting at 1 is strange as a user.

Let's get our users to participate as well. This, too, is user friendliness...
* Choose GitHub reaction emoji "+1": "`2021.0` is good"
* Choose emoji "-1": "`2021.0` is bad, `2021.1` is good"

Reply:
Well, let's get this from the table, start with 0.

Reply:
It seems our users are not interested in versioning (It might be that this place is unnoticeable though) and I'm not sure which they prefer.

So, tentatively, I would like to start from `0.1.0` for now.
@Lucki Is that OK with you?

Reply:
Semver instead of a date? Yes, this is already blocking us for too long

Reply:
Pushed tag `0.0.1` (8aa88ca) as preparation for this PR and upcoming initial release.